tests/rptest/tests/redpanda_test.py

```diff
@@ -75,7 +75,11 @@ def _create_initial_topics(self):
         config = self.redpanda.security_config()
         user = config.get("sasl_plain_username")
         passwd = config.get("sasl_plain_password")
-        client = KafkaCliTools(self.redpanda, user=user, passwd=passwd)
+        client = self._client
+
+        if user and passwd:
```

Comment:
Bit weird to have username+password be mutually exclusive with SI topic options.  This will probably come back to bite us at some point, e.g. if we ever change to enabling authentication by default for all tests.

Maybe do a like-for-like replacement here, and just construct a RpkTool with username+password?